src/components/Touchable.js

```diff
@@ -12,17 +12,20 @@ const defaultHitSlop = {
   bottom: 16,
 };
 
+type Props = {
+  // when on press returns a promise,
+  // the button will toggle in a pending state and
+  // will wait the promise to complete before enabling the button again
+  // it also displays a spinner if it takes more than WAIT_TIME_BEFORE_SPINNER
+  onPress: ?() => ?Promise<any>,
+  children: *,
+  event: string,
+  eventProperties?: Object,
```

Comment:
Object will be a problem when we migrate from Flow to TS. We should replace it with `{[key: string]: any}` when we can.